clients/src/main/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumer.java

```diff
@@ -150,11 +150,9 @@ public class LegacyKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
             LogContext logContext = createLogContext(config, groupRebalanceConfig);
             this.log = logContext.logger(getClass());
             boolean enableAutoCommit = config.getBoolean(ENABLE_AUTO_COMMIT_CONFIG);
-            groupId.ifPresent(groupIdStr -> {
-                if (groupIdStr.isEmpty()) {
-                    log.warn("Support for using the empty group id by consumers is deprecated and will be removed in the next major release.");
-                }
-            });
+
+            if (this.groupId.isPresent() && this.groupId.get().isEmpty())
```

Comment:
Trim please.

Reply:
The use of `AbstractConfig.getString()` trims the returned value, so it will be sans whitespace already. A newly-added unit test shows this to be the case.

Reply:
I was wondering.